### DISCUSSION TITLE
gql-tag-operations-preset and babel plugin doesn't give working fragments

### DISCUSSION THREAD

**Discussion by Cordazar** [dotansimha/graphql-code-generator#8811] [Q&A]
I've been trying to use `@graphql-codegen/gql-tag-operations-preset`  but got issues with unknown fragments when running queries. The documentation says to use the BabelPlugin for this but it doesn't change anything for me, I still get same error. Any ideas as to what I should try next?

From my vite config:
```ts
    plugins: [
        react({
            babel: {
                plugins: [[babelPlugin, {artifactDirectory: './src/api/types/__generated__'}]]
            }
        }),
    ]
```

From my codegen.ts:
```ts
const config: CodegenConfig = {
    schema: './schema.json',
    documents: ['src/**/!(*.test).{tsx,ts,js,jsx}'],
    generates: {
        './src/api/types/__generated__': {
            plugins: [],
            preset: 'gql-tag-operations-preset',
            presetConfig: {
                augmentedModuleName: 'graphql-tag'
            }
        }
    },
    ignoreNoDocuments: true,
    hooks: {
        afterAllFileWrite: 'prettier --write src/api/types/__generated__'
    }
};
```

**Comment by saihaj** [comment]
Can you share a [minimal reproduction](https://en.wikipedia.org/wiki/Minimal_working_example). You can either use [StackBlitz](https://stackblitz.com/), [CodeSandbox](https://codesandbox.io/) or a public GitHub repo.

**Comment by MH4GF** [comment]
It is a different preset, but I think it is close. I have confirmed that bundle size can be reduced by setting up babelPlugin. The source code is also included, which may be helpful.
https://github.com/dotansimha/graphql-code-generator/discussions/8829
